### ISSUE TITLE
Medusa is killed with OOM after 4h of fuzzing

### ISSUE THREAD

**Issue by aviggiano** [crytic/medusa#237]
Hello,

I am running Medusa on a `c5.large` instance (4 GiB memory) and it was killed by the OOM killer.

I supposed it might be incurring some issues related to memory management.

```
yarn run v1.19.2
$ medusa fuzz --corpus-dir ../../medusa
[1m[32m⇾[0m[0m Reading the configuration file at: [1m.../packages/contracts/medusa.json[0m
[1m[32m⇾[0m[0m Compiling targets with [1mcrytic-compile[0m
[1m[32m⇾[0m[0m Creating [1m10[0m workers...
[1m[32m⇾[0m[0m [1mfuzz: [0melapsed: [1m0s[0m, calls: [1m0 (0/sec)[0m, seq/s: [1m0[0m, 
// ...
[1m[32m⇾[0m[0m [1mfuzz: [0melapsed: [1m4h32m30s[0m, calls: [1m1858278 (46/sec)[0m, seq/s: [1m0[0m, coverage: [1m121[0m
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
[Wed Oct  4 00:38:58 2023] Out of memory: Killed process 1436 (medusa) total-vm:14364560kB, anon-rss:3235232kB, file-rss:1664kB, shmem-rss:0kB, UID:1000 pgtables:24808kB oom_score_adj:0
```

**Comment by anishnaik** [comment]
Hey @aviggiano looks like we might have a potential memory leak. Will investigate ASAP

**Comment by aviggiano** [comment]
Btw, I am running a [fork](https://github.com/aviggiano/medusa) of https://github.com/crytic/medusa `master` with `dev/merge-assertion-and-property-mode` and `dev/no-multi-abi`

**Comment by anishnaik** [comment]
Closed by #290
